compiler-rt/lib/ctx_profile/CtxInstrContextNode.h

```diff
@@ -114,9 +114,14 @@ class ContextNode final {
 };
 
 /// Abstraction for the parameter passed to `__llvm_ctx_profile_fetch`.
+/// `startContextSection` is called before any context roots are sent for
+/// writing. Then one or more `writeContextual` calls are made; finally,
+/// `endContextSection` is called.
 class ProfileWriter {
 public:
+  virtual void startContextSection() = 0;
```

Comment:
You could have this method return an object which runs `endContextSection` on destruction. Then we wouldn't have to worry about pairing up the calls to start and end each time. Perhaps it's too much for a simple interface like this though.

Reply:
I thought of that but yes, seemed overkill. It's a low-level interface, really meant to go between compiler-rt and whatever service handles writing the profile.